src/execution_context.rs

```diff
@@ -23,13 +26,18 @@ impl<TS: TypeSystem> ExecutionContext<TS> {
             frame: 0,
             return_value: Default::default(),
             right_operand: Default::default(),
+            popped_value: Default::default(),
```

Comment:
Why do we have a new register for this instead of using the return register?

Reply:
because this isnt a real cpu so we arent really limited in the same way so I figured this might add some flexibility in how it is used in other projects, not just fender
now that you pointed it out I relized that this is curently stupid cause you cant access that register with the commands so I will switch it to return but I think we should consider adding this in as like an optional feature

Reply:
Yeah that's sort of my main issue with adding more registers, we just end up with a lot more register-specific instructions. If we wanted, we could switch to a sized array holding all registers so we wouldn't need unique instructions for each of them, but that also has the effect of obscuring the purpose of each register. Not that our current registers are perfect, since one of them has a double/triple meaning, though. I could go either way.

Reply:
I was actually thinking of that earlier and that it might make more sense 
we have a lot of repeated instructions for similar things. we could do it more like an actual CPU instruction set
we have one vec of registers and then have a series of constants that we use to identify the registers

Reply:
I'd be good with that approach, though I'd ask that you use a const generic usize parameter and make it a sized array, not a vec. Registers should be fast, and if they're in a Vec (on the heap), they'll be no faster than our "stack" (which is also on the heap). Is there a way we could enforce N >= 3?

Reply:
I meant an array there is no reason to use a vec cause it would just be a constant size

Reply:
👍

Reply:
test